src/Pact/Analyze/Translate.hs

```diff
@@ -114,10 +115,13 @@ data TranslateEnv
     { _teInfo           :: Info
     , _teNodeVars       :: Map Node (Text, VarId)
     , _teRecoverability :: Recoverability
+    , _teTestMode       :: IsTest
     }
 
-mkTranslateEnv :: Info -> [Arg] -> TranslateEnv
-mkTranslateEnv info args = TranslateEnv info nodeVars mempty
+data IsTest = IsTest | IsntTest
```

Comment:
Brian, I promised to remove this, but that proved quite a bit more complicated. I'm still open to it if we figure out a simple way.

Reply:
OK cool -- let's talk about this one. It would be nice to remove

Reply:
can inject `genId id` vs `pure 0`. and then have translation impl fit this client-provided action using `zoom`